src/renderer/captureTool/js/capture.js

```diff
@@ -0,0 +1,1014 @@
+            prefSetPayload: prefSetPayload
+        });
+    };
+
+    gpii.captureTool.updateCapturedPreferences = function (that) {
+        var prefs = {};
+        fluid.each(that.model.capturedSettings, function (appData, appId) {
+            if (appData.numberOfSettings > 0) {
+                var appUri = "http://registry.gpii.net/applications/" + appId;
+                prefs[appUri] = {};
+                fluid.each(appData.settings, function (settingVal, settingId) {
+                    prefs[appUri][settingId] = settingVal;
+                });
+            }
+        });
+        that.fullChange("capturedPreferences", prefs);
+    };
+
+    gpii.captureTool.updatePreferencesToKeep = function (that) {
+        var togo = {};
+        fluid.each(that.model.settingsToKeep, function (unparsedSetting) {
+            var appSettingPair = unparsedSetting.split(":");
+            var appId = appSettingPair[0];
+            var appUri = "http://registry.gpii.net/applications/" + appId;
+            var settingId = appSettingPair[1];
+            var settingValue = that.model.capturedPreferences[appUri][settingId];
+            if (!togo[appUri]) {
+                togo[appUri] = {};
+            }
+            togo[appUri][settingId] = settingValue;
+        });
+        that.fullChange("preferencesToKeep", togo);
+    };
+
+    /*
+     * On the settings to keep page, this will be invoked each time an individual
+     * setting is changed, in order to update the heirarchical checkboxs for each
+     * solution, which can be in indeterminate form if only some of the solutions are
+     * checked.
+     */
+    gpii.captureTool.updateSolutionSettingsTree = function (that) {
+        fluid.each(that.model.capturedSettingsToRender, function (solution) {
+            var numSelected = 0;
+            fluid.each(Object.keys(solution.renderSettings), function (settingId) {
+                var solSetting = solution.id + ":" + settingId;
+                if (fluid.contains(that.model.settingsToKeep, solSetting)) {
+                    numSelected += 1;
+                }
+            });
+
+            // sgithens TODO how to use selectors for this
+            var checkboxEl = $("#fc-solution-" + solution.id.replace(/\./g, "\\."))[0];
+            // In case this is called before the page is finished rendering
+            if (!checkboxEl) {
+                return;
+            }
+            if (numSelected === 0) {
+                checkboxEl.indeterminate = false;
+                checkboxEl.checked = false;
+            }
+            else if (solution.numberOfRenderSettings === numSelected) {
+                checkboxEl.indeterminate = false;
+                checkboxEl.checked = true;
+            }
+            else {
+                checkboxEl.indeterminate = true;
+                checkboxEl.checked = false;
+            }
+        });
+    };
+
+    gpii.captureTool.onSolutionHeaderToKeepCheck = function (that, event) {
```

Comment:
I'd refactor these to all just accept the target node as 2nd argument to be clearer that we don't have a dependency on the entire DOM event

Reply:
refactored